### ISSUE TITLE
Doc for setting up test environment (p4c / bmv2 / mininet, packet generation, packet capture)

### ISSUE THREAD

**Issue by r12f** [futurewei-cloud/alioth-app#4]

**Comment by ZER0-Nu1L** [comment]
@r12f I tried p4app(https://github.com/p4lang/p4app) before, it seemed that it is much more convenient for us to develop a prototype. 🤔

**Comment by r12f** [comment]
Yep, for simply writing p4 programs, p4app is great, feel free to add it in!

I set it up this way for us, because it allows us to easily change the behavior model code or flavor without rebuilding a new p4app container. This is for things like debugging bmv2 with gdb, exploring more on internal structures, or enabling debugger or certain features. Since we are building the language, having these ability for mininet is important. (btw, I have found an small issue in bmv2 and currently close to be merged. And with p4app, everything is too nicely wrapped up, so we are actually losing certain flexibilities)

Also, it only contains bmv2 as backend, so it is better to a more unified way to call other compilers, like Tofino. So it is better to add makefile shortcuts for running frequently used commands. For examples: 
- `just p4app` which calls p4app with all the setups that we use.
- `just p4cbf` which calls bf-p4c with all the setups that we use.

Besides above, it doesn't provides a way to generate test traffic, if my understanding is correct. But to test p4 programs, simply ping might not be enough. So we will need a doc / script for it.

**Comment by r12f** [comment]
This is the PR I mentioned btw: https://github.com/p4lang/behavioral-model/pull/1174. It should be close to be merged, since already got approved but might need some additional thing to pass the last gate.
